### PR TITLE
Introduce ChannelCollectionType

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Form/Type/ChannelCollectionType.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                ], $options['entry_options'], [
+                    'block_name' => 'entry',
+                    'channel' => $channel,
+                ]));
+            }
+        });
+    }
+
+    /**
+     * @{@inheritdoc}
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Form/Type/ChannelCollectionType.php]: `@{@inheritdoc}` - `{@inheritdoc}`

File: src/Sylius/Bundle/CoreBundle/Form/Type/Product/ChannelPricingType.php [hunk 2]
```diff
@@ -26,44 +29,26 @@
...
-                ]);
-            })
+            ->add('price', MoneyType::class, [
+                'label' => $options['channel']->getName(),
+                'currency' => $options['channel']->getBaseCurrency()->getCode(),
+            ])
         ;
+
+        $builder->addEventListener(FormEvents::SUBMIT, function (FormEvent $event) use ($options) {
+            /** @var ChannelPricing|null $channelPricing */
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Form/Type/Product/ChannelPricingType.php]: I suppose this variable should never be `null`.

File: src/Sylius/Bundle/CoreBundle/Form/Type/ChannelCollectionType.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            'entry_options' => [],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
+    {
+        return 'collection';
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Form/Type/ChannelCollectionType.php]: Why not `channel_collection`?

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Form/Type/ChannelCollectionType.php]: To reuse collection templates (it does not extend `CollectionType`, but it behaves like it (with options `allow_add`, `allow_delete`, `delete_empty` set to false).

**Comment by pamil** [comment]
In fact it does not include a BC break as `ChannelPricingsType` was introduced just a few days ago.

**Comment by pjedrzejewski** [comment]
Thanks Kamil! Looks good!

**Pull request merged**